### ISSUE TITLE
Misleading handling when the config file is not compliant with the schema

### ISSUE THREAD

**Issue by rsemlal-tas** [hadolint/hadolint#816]
- [x] This is a bug report
- [ ] This is a feature request
- [X] I searched existing issues before opening this one

When a malformed config file is passed to hadlint, it prints a "non-escaped" error message and exits with a `0 ` exit code (as long as there are no issues with the scanned Dockerfile)

### Steps to reproduce the behavior

**Output of `hadolint --version`**:
```bash
$ hadolint --version
Haskell Dockerfile Linter 2.10.0
```

Steps to reproduce:

```bash
$ echo 'FROM ubuntu:18.04' > Dockerfile # Any Dockerfile will do
$ echo 'gibberish' > config # Create a config file that will NOT pass the schema validation

$ hadolint -c config Dockerfile # Use the malformed config file
"Error parsing your config file in  'config':\nIt should contain one of the keys 'override', 'ignored'\nor 'trustedRegistries'. For example:\n\nignored:\n\t- DL3000\n\t- SC1099\n\n\nThe key 'override' should contain only lists with names 'error',\n'warning', 'info' or 'style', which each name rules to override the\nseverity on. For example:\n\noverride:\n\terror:\n\t\t- DL3008\n\n\nThe key 'trustedRegistries' should contain the names of the allowed\ndocker registries:\n\ntrustedRegistries:\n\t- docker.io\n\t- my-company.com\n\nexpected Configuration instead of !!str\n"

$ echo $?
0
```

### Expected behavior

An understandable error message should explain that the configuration file is malformed and the command should exit with a non zero exit code.

### Actual behavior

The error message on the output is printed with escae characters, and the exit code is 0

```bash
$ hadolint -c config Dockerfile # Use the malformed config file
"Error parsing your config file in  'config':\nIt should contain one of the keys 'override', 'ignored'\nor 'trustedRegistries'. For example:\n\nignored:\n\t- DL3000\n\t- SC1099\n\n\nThe key 'override' should contain only lists with names 'error',\n'warning', 'info' or 'style', which each name rules to override the\nseverity on. For example:\n\noverride:\n\terror:\n\t\t- DL3008\n\n\nThe key 'trustedRegistries' should contain the names of the allowed\ndocker registries:\n\ntrustedRegistries:\n\t- docker.io\n\t- my-company.com\n\nexpected Configuration instead of !!str\n"

$ echo $?
0
```

When using hadolint in a ci pipeline, the problem is that when there is a syntax error in the config file, the failure is silent and hadlint does its check with default values. This means that users will have to explicitely check the ci jobs output to detect issues with the config file, if they don't the issue will not be detected immediately

**Comment by lorenzo** [comment]
would you like to send a PR to fix this? Adding the exit code should be trivial
